addons/test_mail/tests/test_mail_activity.py

```diff
@@ -1001,3 +1001,54 @@ def test_week_grouping(self):
         self.assertEqual(groups[0][groupby], pg_groups["overdue"])
         self.assertEqual(groups[1][groupby], pg_groups["today"])
         self.assertEqual(groups[2][groupby], pg_groups["planned"])
+
+
+class TestTours(HttpCase):
+    def test_activity_view_data_with_offset(self):
```

Comment:
As it's not obvious at first glance that the fetched record are ordered in one order (date desc) and on the page in the reverse order (date_deadline asc), I would add a small docstring here or in the js counter part.